### ISSUE TITLE
Highlighting gets stuck when using "Add Selection To Next Find Match"

### ISSUE THREAD

**Issue by pubfnbar** [microsoft/vscode#149565]
Issue Type: <b>Bug</b>

Steps to Reproduce:
1. Create a new text file.
2. Copy and paste the following (without the quotes) to the new file: "asdf qwer asdf zxcv asdf qwer"
3. Place the cursor either inside, right after, or right before either one of the "asdf" words or one of the "qwer" words (make sure that no text is selected).
4. Run the command "Add Selection To Next Find Match" either by selecting it from the command palette or by pressing its default shortcut: Command+D 
5. Notice now that the word that was nearest to the cursor is highlighted blue (your color may vary) and all the other occurrences of the same word are also highlighted (with a slightly different color).
6. Move the cursor either by pressing one of the arrow keys (up/down/left/right) or click somewhere in the file where the cursor is not currently at.
7. Notice that all the words that were previously highlighted are still highlighted, and the highlighted regions remain even if you edit the text.

Expected Results:
I expected all the highlightings to disappear after step 6 (when the cursor is moved).

Additional Observations:
* This issue doesn't occur if you first select the word (either "asdf" or "qwer") before you run the command "Add Selection To Next Find Match".
* This issue doesn't occur if you run the command "Add Selection To Next Find Match" consecutively so many times that all the occurrences of the word have been selected before you move the cursor.
* This issue doesn't occur if there's only one occurrence of the word under the cursor (you can test this with the lone word "zxcv").

VS Code version: Code 1.67.1 (da15b6fd3ef856477bf6f4fb29ba1b7af717770d, 2022-05-06T12:37:14.619Z)
OS version: Darwin arm64 21.4.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 3|
|Memory (System)|16.00GB (0.49GB free)|
|Process Argv|--crash-reporter-id 4b2d2c70-87d5-4d6b-a9f6-6a9b2467b824|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none

**Comment by gjsjohnmurray** [comment]
/duplicate of #148893 and will be fixed in upcoming 1.67.2

**Comment by VSCodeTriageBot** [comment]
Thanks for creating this issue! We figured it's covering the same as another one we already have. Thus, we closed this one as a duplicate. You can search for [similar existing issues](https://github.com/microsoft/vscode/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+). See also our [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Happy Coding!
